CMakeLists.txt

```diff
@@ -28,6 +28,33 @@ if(NOT CMAKE_BUILD_TYPE)
   message(STATUS "Build type not specified: Use Release by default.")
 endif()
 
+#------------------------------------------------------------------------
+# Static Windows Runtime
+#   Option to statically link to the Windows runtime. Maybe only 
+#   applies to WIN32/MSVC.
+#------------------------------------------------------------------------
+if (MSVC)
+    option( USE_STATIC_RUNTIME "Set ON to change /MD(DLL) to /MT(static)" OFF )
+    if (USE_STATIC_RUNTIME)
+        set(CompilerFlags
+            CMAKE_CXX_FLAGS
+            CMAKE_CXX_FLAGS_DEBUG
+            CMAKE_CXX_FLAGS_RELEASE
+            CMAKE_C_FLAGS
+            CMAKE_C_FLAGS_DEBUG
+            CMAKE_C_FLAGS_RELEASE
+            )
+        foreach(CompilerFlag ${CompilerFlags})
+            string(REPLACE "/MD" "/MT" ${CompilerFlag} "${${CompilerFlag}}")
+        endforeach()    
+        message(STATUS "Using /MT STATIC runtime")
+    else ()
+        message(STATUS "Using /MD DYNAMIC runtime")
```

Comment:
If I'm not missing something, you just assume that it is `MD` by default, but it doesn't have to be

Reply:
> I think you will find the cmake MSVC generator default is /MD /MDd, unless you instruct otherwise... 

It doesn't matter what the default value is. As you said, any part of the broader build environment can change this value. This message above is hard-coded and doesn't reflect the actual value at runtime. Therefore, the message above is wrong and misleading when the value is not `/MD` and `USE_STATIC_RUNTIME` and is `OFF`

Reply:
@Croydon I see - so what is your suggestion?